dask/array/reductions.py

```diff
@@ -92,7 +93,8 @@ def partial_reduce(func, x, split_every, keepdims=False, dtype=None, name=None):
 
     >>> partial_reduce(np.min, x, {0: 1, 2: 3})    # doctest: +SKIP
     """
-    name = name or 'p_reduce-' + tokenize(func, x, split_every, keepdims, dtype)
+    name = name or funcname(func) + '-' + tokenize(func, x, split_every,
```

Comment:
This means that if the name is provided to `partial_reduce`, then the output keys will have the same name regardless of level in the reduction tree. I'd either always add `tokenize(x)` to the name here, or add it in the loop on in `reduction` above.